tests/generation/test_utils.py

```diff
@@ -421,7 +421,7 @@ def test_greedy_generate(self):
             model = model_class(config).to(torch_device).eval()
             output_generate = self._greedy_generate(model=model, inputs_dict=inputs_dict)
 
-            if model.config.get_text_config(decoder=True).is_encoder_decoder:
+            if model.config.is_encoder_decoder:
```

Comment:
The original `model.config.get_text_config(decoder=True).is_encoder_decoder` was added because of `BLIP2`, which was not setting its outer-level `is_encoder_decoder` attribute correctly. This PR fixes the root issue there.

In practice, this line was not right: if we want to check if a model is an encoder-decoder model, we can't look at it's decoder config only :)